code/game/objects/items/devices/scanners.dm

```diff
@@ -577,6 +577,10 @@ GENE SCANNER
 	var/cooldown_time = 250
 	var/accuracy // 0 is the best accuracy.
 
+/obj/item/analyzer/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/gas_analyzer, TRUE)
```

Comment:
This `TRUE` argument isn't used, and isn't necessary

Reply:
right, its because i changed visible to be on the signal

Reply:
well, now its used again.